symengine/rational.h

```diff
@@ -236,9 +239,12 @@ void get_num_den(const Rational &rat,
         const Ptr<RCP<const Integer>> &den);
 
 //! convenience creator from two longs
-inline RCP<const Number> rational(long n, long d) {
-    return Rational::from_two_ints(n, d);
-    }
+inline RCP<const Rational> rational(long n, long d)
+{
+    mpq_class rat(n, d);
+    rat.canonicalize();
+    return make_rcp<const Rational>(rat);
+}
```

Comment:
I am not sure if I understand your question, but the Rational constructor requires the numbers to actually be a rational, not an integer, see here:

https://github.com/symengine/symengine/blob/63e9d3fe44827576d9afe839d892974bb5fbb600/symengine/rational.cpp#L5

and the `is_canonical` function. Since in your new code you construct an `mpq_class` even if it is an integer, then it must fail in Debug mode in `is_canonical` in that case.